acl/errors.go

```diff
@@ -61,18 +61,67 @@ func IsErrPermissionDenied(err error) bool {
 	return err != nil && strings.Contains(err.Error(), errPermissionDenied)
 }
 
+// Arguably this should be some sort of union type.
+// The usage of Cause and the rest of the fields is entirely disjoint.
```

Comment:
I'm pretty sure we'll be able to make this not entirely disjoint in the future! There are at least 2 places in enterprise where we'll want to use `Cause` along with the other fields. The current implementation doesn't support that yet, but I think that's ok. We can add support when it comes up.